README.md

⚠ This software is developed for a specific server and should not be used for other servers! ⚠
[<img alt="PowerNukkitX" width="838" src="https://raw.githubusercontent.com/PowerNukkitX/PowerNukkitX/master/blob/images/PNX_BANNER.png" />](https://www.powernukkitx.com)


CHANGELOG.md
<<<<<<< HEAD
# 更新日志
本项目所有值得注意的变化都将会记录在这个文件中

摘要：该文档的格式基于[Keep a Changelog](https://keepachangelog.com/en/1.0.0/)二次修改,
并且本项目遵守[Semantic Versioning](https://semver.org/spec/v2.0.0.html)并在主要版本前加上上游的主要版本号，这样我们就能更好的区别于Nukkit 1.X和2.X。

## 目录

1. <a href="#CataLogs-Swlang">🌐 Switch Languages / 切换语言 </a>
2. <a href="#CataLogs-Join-the-community">💬 Join the Community / 加入我们 </a>
3. <a href="#CataLogs-Version-history">🔖 Version history / 历史版本 </a>

## [1.19.21-r4-PNX](https://github.com/PowerNukkitX/PowerNukkitX/actions) - Dev
该版本目前支持了Minecraft:BE `1.19.21 (协议版本545)`.

## 新增内容

- [#613] 支持Deep Dark“幽暗”群系（阶段#1）。

## 修改记录

- [#620] 移除`StringArrayTag.java`。
- [#621] 默认配置中添加反矿透配置示例。
- [#623] 正确的anti-xray配置。

## BUG修复

- [#612] 修复与NPC插件的内容。
- [#617] 修复漏斗。

## [1.19.21-r3-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.21-r3) - 2022-9-2
该版本目前支持了Minecraft:BE `1.19.21 (协议版本545)`.

## 新增内容

- [#610] 实现矿车（InventoryHolder）+漏斗。

## 修改记录

- [#599] 重构实体注册。
- [#601] js-java互操作性增强。
- [#602] 反矿物透视改进。
- [#611] 优化玩家移动。

## BUG修复

- [#603] 修复漏斗熔炉刷物品BUG。
- [#605] 修复错误的箭初始速度。
- [#607] fix entity death smoke + potion effect cloud + explosion。
- [#615] 修复/effect命令。

## [1.19.21-r2-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.21-r2) - 2022-8-24
该版本目前支持了Minecraft:BE `1.19.21 (协议版本545)`.

### 新增内容

- [#572] 基本结构生成实现。

### BUG修复

- [#591]/[#592] 修复木板->栅栏门,羊毛->羊毛毯,玻璃->玻璃板的配方（在[#596]中修复）。

## [1.19.21-r1-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.21-r1) - 2022-8-23
该版本目前支持了Minecraft:BE `1.19.21 (协议版本545)`.

### 新增内容

- [#594] 1.19.21-r1正式发布。
- [#587] 添加村民交易接口。 
- [#586] 内置反矿物透视。

### 修改记录

- [#586] 区块发送优化。
- [#593] 实现兼容1.19.21（协议版本545）。

### BUG修复

- [#575] 修复自定义方块变成空气之后不会保存的BUG（在[#585]中修复）。
- [#584] 修复自定义方块的一堆BUG。

## [1.19.20-r5-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.20-r5) - 2022-8-18
该版本目前支持了Minecraft:BE `1.19.20 (协议版本544)`.

### 新增内容

- [#576] 1.19.20-r5正式发布。
- [#571] 添加初步能源系统实现。
- [#574] 添加shaded警告。

### 修改记录

- [#537] 完善自定义方块。
- [#550] 完善配方。
- [#562] 通过Module获取资源文件而不是通过ClassLoader。
- [#564] 优化Terra内存占用。

### BUG修复

- [#552] 修复进入地狱客户端崩溃的BUG。
- [#554] 修复chunkSectionCount无法写入区块nbt的BUG。
- [#556] 修复红树树叶的状态BUG。
- [#557] 修复Teera内存溢出的漏洞。
- [#563] 修复竹子可以被活塞推动的BUG。
- [#565] 修复3D生物群系读写
- [#568] 修复铁砧在含水方块上无限掉落的BUG。
- [#569] 修复实体y<0时的异常伤害。
- [#570] 修复杜鹃花掉落概率。
- [#573] 修复熔炉配方。


## [1.19.20-r4-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.20-r4) - 2022-8-13 -（Pre Release）
该版本目前支持了Minecraft:BE `1.19.20 (协议版本544)`.

### 新增内容

- [#542] 1.19.20-r4（Pre Release）版发布。
- [#536] Chunk中新的getMaxHeight和getMinHeight方法。

### 修改记录

- [#542] 将terra版本更新至6.2.0-Release。

### BUG修复

- [#536] 修复Chunk中维度相关方法NPE。

## [1.19.20-r3-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.20-r3) - 2022-8-13
该版本目前支持了Minecraft:BE `1.19.20 (协议版本544)`.

### 新增内容

- [#524] `支持3D生物群系`和自定义维度API（TODO）。

### 修改记录

- [#524] 更改了Anvil格式的读写方法以提高性能。

### BUG修复

- [#427] 修复`"this.skyLight" is null`漏洞（在[#524]中修复）。
- [#520] 修复在termux中无法启动PowerNukkitX的问题（在[#532]中修复）。

## [1.19.20-r2-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.20-r2) - 2022-8-12
该版本目前支持了Minecraft:BE `1.19.20 (协议版本544)`.

### 修改记录

- [#519] 新/execute命令格式。
- [#523] 优化JS插件与ava互调用。

### BUG修复

- [#525] 修复杜鹃树叶不消失的BUG（在[#528]中修复）。
- [#526] 物品不能着色（在[#527]中修复）。

## [1.19.20-r1-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.20-r1) - 2022-8-10 - (Pre Release)
该版本目前支持了Minecraft:BE `1.19.20 (协议版本544)`.

### 修改记录

- [#515] 实现兼容1.19.20（协议版本544）。

### BUG修复

- [#511] 修复无法种植大型云杉树的问题。
- [#512] 修复修复与jar-in-jar多级插件间的兼容。
- [#514] 修复当玩家在坐骑时速度过快会被误检为瞬移的问题。

## [1.19.10-r1-PNX](https://github.com/PowerNukkitX/PowerNukkitX/releases/tag/1.19.10-r1) - 2022-8-7
该版本目前支持了Minecraft:BE `1.19.10 (协议版本534)`.

### 新增内容

- [#510] 1.19.10-r1正式发布。

### 修改记录

- [#506] 1.19.10-r1版本更新。

## [1.6.0.0-PNX](https://github.com/PowerNukkitX/PowerNukkitX/actions/runs/2808051758) - EOL - (Dev)
该版本目前支持了Minecraft:BE `1.19.10 (协议版本534)`.

### 新增内容

- [#17] 在PowerNukkitX内实现384限高世界（目前仅限主世界）。
- [#45] 添加1.18.10版本中新增的物品。
- [#106] 在PowerNukkitX内初步实现地图。
- [#146] 为PowerNukkitX添加了Bootstrap启动器，可无需安装Java17即可使用PowerNukkitX。
- [#161] 在PowerNukkitX内实现并内置了香草命令和命令方块。
- [#177] 在PowerNukkitX内添加Terra生成器（[FAQ](https://doc.powernukkitx.cn/zh-cn/faq/Terra%E9%97%AE%E9%A2%98.html)）。
- [#236] 在PowerNukkitX中初步实现并完成了JavaScript插件支持[（点此查看开发文档）](https://doc.powernukkitx.cn/zh-cn/plugin-dev/js/%E6%A6%82%E8%BF%B0.html)。
- [#288] 实现/summon命令。`（Tips：会和MobPlugin的/summon命令产生冲突）`
- [#307] 实现/function命令。
- [#326] 实现RAWTEXT（/tellraw /titleraw）。
- [#352] 实现NPC-API和实现NPC功能。
- [#354] 实现自定义方块 / 物品 / 实体（完善中，[文档](https://doc.powernukkitx.cn)待补充，自定义实体在[#458]中实现）。
- [#363] 实现NPC SKIN切换。
- [#365] 添加toSnbt。
- [#370] 添加JS跨插件互操作。
- [#384] 实现swift_sneak附魔效果。
- [#385] 实现darkness药水效果。
- [#387] 支持新版成就界面。
- [#389] 实现潜声方块实体。
- [#414] 实现细雪方块。
- [#416] 添加`PlayerFreezeEvent`事件。
- [#425] 初步完成`JS Feature`架构。
- [#431] 新的生物AI（基本架构）。
- [#433] 实现常加载区块以及对应命令。
- [#426] 为字节码调用失败的事件添加反射逃生门。
- [#446] 实现DeathInfo。
- [#468] 添加`ServerStartedEvent`事件。
- [#470] 添加OtherSide唱片。
- [#481] 实现猪牛鸡的生物AI。
- [#483] 实现僵尸的生物AI。
- [#492] 添加SNBT反序列化。
- [#494] 新增箱船（Chest boat）。
- [#500] 实现苦力怕生物AI。

### 修改记录

- [#45] 实现兼容1.18.10（协议版本486）。
- [#78] 将新增加的物品添加创造物品栏中。
- [#132] 将whitelist更改为allowlist。
- [#243] 实现兼容1.18.30 (协议版本503)。
- [#275] 实现基本的实体运动处理。
- [#330] 完善目标选择器。
- [#333] 初步实现Mob的Equipment。
- [#337] 改进生物Inventory。
- [#346] 更新饥饿值计算。
- [#359] 修改配方。
- [#366] 更新terra版本。
- [#367] 完善NPC接口。
- [#368] 完善NPC提示框。
- [#373] 将terra版本更新至6.0.0-Release。
- [#375] 实现滚动字幕API。
- [#380] 实现兼容1.19.0 (协议版本527)。
- [#390] 支持带有_的玩家名称解析。
- [#402] 合并NukkitX的修改。
- [#411] 优化/version命令。
- [#418] 优化事件调用性能。
- [#428] NPC Dialog协议逻辑同步1.19.0。
- [#443] 完善Mapping。
- [#445] 实现兼容1.19.10 (协议版本534)。
- [#455] 更新资源文件。
- [#461] 更新启动命令检测 + 弃用submodule。
- [#466] `修改自定义方块API。`
- [#467] 调整Version命令更新检查。
- [#473] 改进status命令+更新依赖库。
- [#477] 增强JS引擎的自定义性和兼容性。
- [#489] 优化寻路逻辑。
- [#490] 更改boss实体位置以适配384高度。
- [#491] 改进实体AI。

src/main/java/cn/nukkit/item/ItemID.java
package cn.nukkit.item;

import cn.nukkit.api.PowerNukkitOnly;
import cn.nukkit.api.PowerNukkitXOnly;
import cn.nukkit.api.Since;

import static cn.nukkit.utils.Utils.dynamic;

public interface ItemID {
    @PowerNukkitOnly @Since("FUTURE") int STRING_IDENTIFIED_ITEM = dynamic(255);
    int IRON_SHOVEL = 256;
    int IRON_PICKAXE = 257;
    int IRON_AXE = 258;
    int FLINT_STEEL = 259;
    int FLINT_AND_STEEL = 259;
    int APPLE = 260;
    int BOW = 261;
    int ARROW = 262;
    int COAL = 263;
    int DIAMOND = 264;
    int IRON_INGOT = 265;
    int GOLD_INGOT = 266;
    int IRON_SWORD = 267;
    int WOODEN_SWORD = 268;
    int WOODEN_SHOVEL = 269;
    int WOODEN_PICKAXE = 270;
    int WOODEN_AXE = 271;
    int STONE_SWORD = 272;
    int STONE_SHOVEL = 273;
    int STONE_PICKAXE = 274;
    int STONE_AXE = 275;
    int DIAMOND_SWORD = 276;
    int DIAMOND_SHOVEL = 277;
    int DIAMOND_PICKAXE = 278;
    int DIAMOND_AXE = 279;
    int STICK = 280;
    int STICKS = 280;
    int BOWL = 281;
    int MUSHROOM_STEW = 282;
    int GOLD_SWORD = 283;
    int GOLDEN_SWORD = 283;
    int GOLD_SHOVEL = 284;
    int GOLDEN_SHOVEL = 284;
    int GOLD_PICKAXE = 285;
    int GOLDEN_PICKAXE = 285;
    int GOLD_AXE = 286;
    int GOLDEN_AXE = 286;
    int STRING = 287;
    int FEATHER = 288;
    int GUNPOWDER = 289;
    int WOODEN_HOE = 290;
    int STONE_HOE = 291;
    int IRON_HOE = 292;
    int DIAMOND_HOE = 293;
    int GOLD_HOE = 294;
    int GOLDEN_HOE = 294;
    int SEEDS = 295;
    int WHEAT_SEEDS = 295;
    int WHEAT = 296;
    int BREAD = 297;
    int LEATHER_CAP = 298;
    int LEATHER_TUNIC = 299;
    int LEATHER_PANTS = 300;
    int LEATHER_BOOTS = 301;
    int CHAIN_HELMET = 302;
    int CHAIN_CHESTPLATE = 303;
    int CHAIN_LEGGINGS = 304;
    int CHAIN_BOOTS = 305;
    int IRON_HELMET = 306;
    int IRON_CHESTPLATE = 307;
    int IRON_LEGGINGS = 308;
    int IRON_BOOTS = 309;
    int DIAMOND_HELMET = 310;
    int DIAMOND_CHESTPLATE = 311;
    int DIAMOND_LEGGINGS = 312;
    int DIAMOND_BOOTS = 313;
    int GOLD_HELMET = 314;
    int GOLD_CHESTPLATE = 315;
    int GOLD_LEGGINGS = 316;
    int GOLD_BOOTS = 317;
    int FLINT = 318;
    int RAW_PORKCHOP = 319;
    int COOKED_PORKCHOP = 320;
    int PAINTING = 321;
    int GOLDEN_APPLE = 322;
    int SIGN = 323;
    int WOODEN_DOOR = 324;
    int BUCKET = 325;

    int MINECART = 328;
    int SADDLE = 329;
    int IRON_DOOR = 330;
    int REDSTONE = 331;
    int REDSTONE_DUST = 331;
    int SNOWBALL = 332;
    int BOAT = 333;
    int LEATHER = 334;
    int KELP = 335;
    int BRICK = 336;
    int CLAY = 337;
    int SUGARCANE = 338;
    int SUGAR_CANE = 338;
    int SUGAR_CANES = 338;
    int PAPER = 339;
    int BOOK = 340;
    int SLIMEBALL = 341;
    int MINECART_WITH_CHEST = 342;

    int EGG = 344;
    int COMPASS = 345;
    int FISHING_ROD = 346;
    int CLOCK = 347;
    int GLOWSTONE_DUST = 348;
    int RAW_FISH = 349;
    int COOKED_FISH = 350;
    int DYE = 351;
    int BONE = 352;
    int SUGAR = 353;
    int CAKE = 354;
    int BED = 355;
    int REPEATER = 356;
    int COOKIE = 357;
    int MAP = 358;
    int SHEARS = 359;
    int MELON = 360;
    int MELON_SLICE = 360;
    int PUMPKIN_SEEDS = 361;
    int MELON_SEEDS = 362;
    int RAW_BEEF = 363;
    int STEAK = 364;
    int COOKED_BEEF = 364;
    int RAW_CHICKEN = 365;
    int COOKED_CHICKEN = 366;
    int ROTTEN_FLESH = 367;
    int ENDER_PEARL = 368;
    int BLAZE_ROD = 369;
    int GHAST_TEAR = 370;
    int GOLD_NUGGET = 371;
    int GOLDEN_NUGGET = 371;
    int NETHER_WART = 372;
    int POTION = 373;
    int GLASS_BOTTLE = 374;
    int BOTTLE = 374;
    int SPIDER_EYE = 375;
    int FERMENTED_SPIDER_EYE = 376;
    int BLAZE_POWDER = 377;
    int MAGMA_CREAM = 378;
    int BREWING_STAND = 379;
    int BREWING = 379;
    int CAULDRON = 380;
    int ENDER_EYE = 381;
    int GLISTERING_MELON = 382;
    int SPAWN_EGG = 383;
    int EXPERIENCE_BOTTLE = 384;
    int FIRE_CHARGE = 385;
    int BOOK_AND_QUILL = 386;
    int WRITTEN_BOOK = 387;
    int EMERALD = 388;
    int ITEM_FRAME = 389;
    int FLOWER_POT = 390;
    int CARROT = 391;
    int CARROTS = 391;
    int POTATO = 392;
    int POTATOES = 392;
    int BAKED_POTATO = 393;
    int BAKED_POTATOES = 393;
    int POISONOUS_POTATO = 394;
    int EMPTY_MAP = 395;
    int GOLDEN_CARROT = 396;
    int SKULL = 397;
    int CARROT_ON_A_STICK = 398;
    int NETHER_STAR = 399;
    int PUMPKIN_PIE = 400;
    int FIREWORKS = 401;
    int FIREWORKSCHARGE = 402;
    int ENCHANTED_BOOK = 403;
    int ENCHANT_BOOK = 403;
    int COMPARATOR = 404;
    int NETHER_BRICK = 405;
    int QUARTZ = 406;
    int NETHER_QUARTZ = 406;
    int MINECART_WITH_TNT = 407;
    int MINECART_WITH_HOPPER = 408;
    int PRISMARINE_SHARD = 409;
    int HOPPER = 410;
    int RAW_RABBIT = 411;
    int COOKED_RABBIT = 412;
    int RABBIT_STEW = 413;
    int RABBIT_FOOT = 414;
    int RABBIT_HIDE = 415;
    int LEATHER_HORSE_ARMOR = 416;
    int IRON_HORSE_ARMOR = 417;
    int GOLD_HORSE_ARMOR = 418;
    int DIAMOND_HORSE_ARMOR = 419;
    int LEAD = 420;
    int NAME_TAG = 421;
    int PRISMARINE_CRYSTALS = 422;
    int RAW_MUTTON = 423;
    int COOKED_MUTTON = 424;
    int ARMOR_STAND = 425;
    int END_CRYSTAL = 426;
    int SPRUCE_DOOR = 427;
    int BIRCH_DOOR = 428;
    int JUNGLE_DOOR = 429;
    int ACACIA_DOOR = 430;
    int DARK_OAK_DOOR = 431;
    int CHORUS_FRUIT = 432;
    int POPPED_CHORUS_FRUIT = 433;
    @Since("1.2.1.0-PN") int BANNER_PATTERN = 434;

    int DRAGON_BREATH = 437;
    int SPLASH_POTION = 438;

    int LINGERING_POTION = 441;

    int COMMAND_BLOCK_MINECART = 443;
    int ELYTRA = 444;
    int SHULKER_SHELL = 445;
    int BANNER = 446;
    int MEDICINE = 447;
    int BALLOON = 448;
    int TOTEM = 450;

    int IRON_NUGGET = 452;

    int ICE_BOMB = 453;

    int TRIDENT = 455;

    int BEETROOT = 457;
    int BEETROOT_SEEDS = 458;
    int BEETROOT_SEED = 458;
    int BEETROOT_SOUP = 459;
    int RAW_SALMON = 460;
    int CLOWNFISH = 461;
    int PUFFERFISH = 462;
    int COOKED_SALMON = 463;
    int DRIED_KELP = 464;
    int NAUTILUS_SHELL = 465;
    int GOLDEN_APPLE_ENCHANTED = 466;
    int HEART_OF_THE_SEA = 467;
    int SCUTE = 468;
    int TURTLE_SHELL = 469;
    int PHANTOM_MEMBRANE = 470;
    int CROSSBOW = 471;
    @PowerNukkitOnly int SPRUCE_SIGN = 472;
    @PowerNukkitOnly int BIRCH_SIGN = 473;
    @PowerNukkitOnly int JUNGLE_SIGN = 474;
    @PowerNukkitOnly int ACACIA_SIGN = 475;
    @PowerNukkitOnly int DARKOAK_SIGN = 476;
    @PowerNukkitOnly int DARK_OAK_SIGN = 476;
    int SWEET_BERRIES = 477;

    int COMPOUND = 499;

    int RECORD_13 = 500;
    int RECORD_CAT = 501;
    int RECORD_BLOCKS = 502;
    int RECORD_CHIRP = 503;
    int RECORD_FAR = 504;
    int RECORD_MALL = 505;
    int RECORD_MELLOHI = 506;
    int RECORD_STAL = 507;
    int RECORD_STRAD = 508;
    int RECORD_WARD = 509;
    int RECORD_11 = 510;
    int RECORD_WAIT = 511;

    int SHIELD = 513;

    @Since("1.6.0.0-PNX")
    @PowerNukkitXOnly
    int GLOW_ITEM_FRAME = 623;

    @Since("1.6.0.0-PNX")
    @PowerNukkitXOnly
    int ITEM_MANGROVE_DOOR = 633;

    @Since("1.6.0.0-PNX")
    @PowerNukkitXOnly
    int RECORD_OTHERSIDE = 626;

    @Since("1.6.0.0-PNX")
    @PowerNukkitXOnly
    int MANGROVE_SIGN = 634;

    @Since("1.6.0.0-PNX")
    @PowerNukkitXOnly
    int OAK_CHEST_BOAT = 638;

    @PowerNukkitXOnly
    @Since("1.6.0.0-PNX")
    int BIRCH_CHEST_BOAT = 639;

    @PowerNukkitXOnly
    @Since("1.6.0.0-PNX")
    int JUNGLE_CHEST_BOAT = 640;

    @Since("1.6.0.0-PNX")
    @PowerNukkitXOnly
    int SPRUCE_CHEST_BOAT = 641;

    @PowerNukkitXOnly
    @Since("1.6.0.0-PNX")
    int ACACIA_CHEST_BOAT = 642;

    @PowerNukkitXOnly
    @Since("1.6.0.0-PNX")
    int DARK_OAK_CHEST_BOAT = 643;

    @PowerNukkitXOnly
    @Since("1.6.0.0-PNX")
    int MANGROVE_CHEST_BOAT = 644;

    @PowerNukkitXOnly
    @Since("1.6.0.0-PNX")
    int CHEST_BOAT = 645;

    @Since("1.6.0.0-PNX")
    @PowerNukkitXOnly
    int GLOW_BERRIES = 654;

    @PowerNukkitOnly
    int CAMPFIRE = 720;

    int SUSPICIOUS_STEW = 734;
    int HONEYCOMB = 736;
    int HONEY_BOTTLE = 737;

    @Since("1.4.0.0-PN")
    int LODESTONECOMPASS = 741;
    @Since("1.4.0.0-PN")
    @PowerNukkitOnly
    int LODESTONE_COMPASS = LODESTONECOMPASS;
    @Since("1.4.0.0-PN")
    int NETHERITE_INGOT = 742;
    @Since("1.4.0.0-PN") int NETHERITE_SWORD = 743;
    @Since("1.4.0.0-PN") int NETHERITE_SHOVEL = 744;
    @Since("1.4.0.0-PN") int NETHERITE_PICKAXE = 745;
    @Since("1.4.0.0-PN") int NETHERITE_AXE = 746;
    @Since("1.4.0.0-PN") int NETHERITE_HOE = 747;
    @Since("1.4.0.0-PN") int NETHERITE_HELMET = 748;
    @Since("1.4.0.0-PN") int NETHERITE_CHESTPLATE = 749;
    @Since("1.4.0.0-PN") int NETHERITE_LEGGINGS = 750;
    @Since("1.4.0.0-PN") int NETHERITE_BOOTS = 751;
    @Since("1.4.0.0-PN") int NETHERITE_SCRAP = 752;
    @Since("1.4.0.0-PN") @PowerNukkitOnly int CRIMSON_SIGN = 753;
    @Since("1.4.0.0-PN") @PowerNukkitOnly int WARPED_SIGN = 754;
    @Since("1.4.0.0-PN") @PowerNukkitOnly int CRIMSON_DOOR = 755;
    @Since("1.4.0.0-PN") @PowerNukkitOnly int WARPED_DOOR = 756;
    @Since("1.4.0.0-PN") int WARPED_FUNGUS_ON_A_STICK = 757;
    @Since("1.4.0.0-PN") @PowerNukkitOnly int CHAIN = 758;
    @Since("1.4.0.0-PN") int RECORD_PIGSTEP = 759;
    @Since("1.4.0.0-PN") @PowerNukkitOnly int NETHER_SPROUTS = 760;

    @Since("1.6.0.0-PNX") int AMETHYST_SHARD = 771;
    @Since("FUTURE") int SPYGLASS = 772;

    @Since("1.4.0.0-PN") @PowerNukkitOnly int SOUL_CAMPFIRE = 801;
    //@PowerNukkitOnly int DEBUG_STICK = <Possible:> 802+;
}
